.github/workflows/test.yml

```diff
@@ -12,8 +18,27 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
+      - name: Get permission
+        id: checkAccess
+        uses: actions-cool/check-user-permission@7b90a27f92f3961b368376107661682c441f6103 # v2.3.0
+        with:
+          require: write
+          username: ${{ github.triggering_actor }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Check permission
+        if: steps.checkAccess.outputs.require-result == 'false' && github.actor != 'dependabot[bot]'
```

Comment:
Allows dependabot to pass anyway. For anyone else, check if we have write access from the previous step, otherwise make the workflow fail with a few debug info about why.

Reply:
Cant they just remove this check with pull_request_target and run it anyways?